cleanlab_studio/internal/api/api.py

```diff
@@ -18,6 +23,30 @@
 cleanset_base_url = f"{base_url}/cleansets"
 
 
+def telemetry(func: Callable[..., Any]) -> Callable[..., Any]:
+    """
+    Decorator to send stack trace to backend if an exception is raised
+
+    Can only use on functions whose first arg is api_key
+    """
+
+    @functools.wraps(func)
+    def tracked_func(api_key: str, *args: Any, **kwargs: Any) -> Any:
+        print("using telemetry")
+        try:
+            result = func(api_key, *args, **kwargs)
+            return result
+        except Exception as err:
+            _ = requests.post(
+                f"{cli_base_url}/telemetry",
+                data=traceback.format_exc(),
+                headers=_construct_headers(api_key),
+            )
```

Comment:
we don't want failed telemetry to ever result in an error visible to users, so let's make sure that this is wrapped in a try-except. one example I can think of is if the user is having network issues. their first request will error out, then they'll see a network error raised from a telemetry request. this would be a bit confusing to users -- best to avoid that

Reply:
ah thanks that's a good point.

Fixed this, and also added function name and args to be sent to backend

Reply:
if this looks good now, I'll add it to all the functions that make api calls to our backend